src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -203,6 +203,11 @@ private static void BeforeExecuteCommand(TypeCharCommandArgs args, CommandExecut
             }
         }
 
+        private static bool CanHaveSemicolon(SyntaxNode currentNode)
+            => currentNode.IsKind(SyntaxKind.FieldDeclaration, SyntaxKind.DelegateDeclaration, SyntaxKind.ArrowExpressionClause)
+                || currentNode is RecordDeclarationSyntax { OpenBraceToken: { IsMissing: true } }
+                || (currentNode is MethodDeclarationSyntax method && method.Modifiers.Any(SyntaxKind.AbstractKeyword));
```

Comment:
also extern methods.  please break this large expression up into multiple statements though.  thanks!

Reply:
also methods in interfaces.